packages/rum-core/test/common/config-service.spec.js

```diff
@@ -126,21 +126,27 @@ describe('ConfigService', function() {
     expect(userContext).toEqual({})
   })
 
-  it('should check config validity', function() {
-    var result = configService.isValid()
-    expect(result).toBe(false)
-
-    configService.setConfig({ serviceName: 'serviceName' })
-    result = configService.isValid()
-    expect(result).toBe(true)
-
-    configService.setConfig({ serverUrl: undefined })
-    result = configService.isValid()
-    expect(result).toBe(false)
+  it('should validate required config options', () => {
+    // Valid
+    const errors1 = configService.validate({ serviceName: 'name' })
+    expect(errors1.length).toEqual(0)
+
+    // Invalid required key serviceName
+    const errors2 = configService.validate({ serviceName: undefined })
+    expect(errors2).toEqual(['Missing serviceName'])
+
+    // Invalid required key serviceName & serverUrl
+    const errors3 = configService.validate({
+      serviceName: undefined,
+      serverUrl: ''
+    })
+    expect(errors3).toEqual(['Missing serviceName', 'serverUrl'])
 
-    configService.setConfig({ serverUrl: 'test' })
-    result = configService.isValid()
-    expect(result).toBe(true)
+    // Invalid characters in serviceName
+    const errors4 = configService.validate({ serviceName: 'abc.def' })
+    expect(errors4).toEqual([
+      'serviceName abc.def contains invalid characters! (allowed: a-z, A-Z, 0-9, _, -, <space>)'
```

Comment:
Maybe something like `serviceName "abc.def" contains invalid characters` would be better!